src/index.js

```diff
@@ -71,19 +96,26 @@ class FloodSub extends BaseProtocol {
    * @param {string} idB58Str peer id string in base58
    * @param {Connection} conn connection
    * @param {PeerInfo} peer peer info
-   * @returns {undefined}
+   * @returns {void}
    *
    */
-  _processConnection (idB58Str, conn, peer) {
-    pull(
-      conn,
-      lp.decode(),
-      pull.map((data) => message.rpc.RPC.decode(data)),
-      pull.drain(
-        (rpc) => this._onRpc(idB58Str, rpc),
-        (err) => this._onConnectionEnd(idB58Str, peer, err)
+  async _processMessages (idB58Str, conn, peer) {
+    const onRpcFunc = this._onRpc
+    try {
+      await pipe(
+        conn,
+        lp.decode(),
+        async function collect (source) {
```

Comment:
The name isn't needing and this isn't collecting anything, so just remove it.

```suggestion
        async function (source) {
```